ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/StandaloneExecutorTest.java

```diff
@@ -40,54 +55,125 @@
+    EasyMock.expect(engine.buildMultipleQueries(eq("CTAS"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(ctasQueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(ctasQuery, ksqlConfig))
+        .andReturn(ctasQueryMetadata).once();
+
+    final QueryMetadata csas2QueryMetadata = EasyMock.niceMock(PersistentQueryMetadata.class);
+    EasyMock.expect(csas2QueryMetadata.getDataSourceType()).andReturn(DataSourceType.KSTREAM);
+    EasyMock.expect(engine.buildMultipleQueries(eq("CSAS2"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(csas2QueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(csas2Query, ksqlConfig))
+        .andReturn(csas1QueryMetadata).once();
+
+    EasyMock.replay(setProperty);
+    EasyMock.replay(csas1Query);
+    EasyMock.replay(createStreamAsSelect1);
+    EasyMock.replay(csas1QueryMetadata);
+    EasyMock.replay(ctasQuery);
+    EasyMock.replay(createTableAsSelect);
+    EasyMock.replay(ctasQueryMetadata);
+    EasyMock.replay(csas2Query);
+    EasyMock.replay(createStreamAsSelect2);
+    EasyMock.replay(csas2QueryMetadata);
+    EasyMock.replay(engine);
+    standaloneExecutor.start();
     EasyMock.verify(engine);
   }
 
-  @Test
-  public void shouldExecutePersistentQueries() {
-    final PersistentQueryMetadata query = EasyMock.niceMock(PersistentQueryMetadata.class);
-    EasyMock.expect(engine.createQueries(anyString(), eq(ksqlConfig)))
-        .andReturn(Collections.singletonList(query));
-    query.start();
-    EasyMock.expectLastCall();
-    EasyMock.replay(query, engine);
-
-    executor.start();
-
-    EasyMock.verify(query);
-  }
+  @Test(expected = KsqlException.class)
+  public void shouldFailIfCsasIsInvalid() {
+    final StandaloneExecutor standaloneExecutor =
+        new StandaloneExecutor(ksqlConfig, engine, queriesFile, udfLoader);
+
+    final List<Pair<String, Statement>> statementPairs = new ArrayList<>();
+    final MetaStore metaStore = EasyMock.niceMock(MetaStore.class);
+    statementPairs.add(new Pair<>("CS", EasyMock.niceMock(CreateStream.class)));
+    statementPairs.add(new Pair<>("CT", EasyMock.niceMock(CreateTable.class)));
+
+    final CreateStreamAsSelect createStreamAsSelect1 = EasyMock.niceMock(CreateStreamAsSelect.class);
+    final Query csas1Query = EasyMock.niceMock(Query.class);
+    EasyMock.expect(createStreamAsSelect1.getQuery()).andReturn(csas1Query);
+    statementPairs.add(new Pair<>("CSAS1", createStreamAsSelect1));
+
+    EasyMock.expect(engine.parseStatements(anyString(), anyObject())).andReturn(statementPairs);
+    EasyMock.expect(engine.getMetaStore()).andReturn(metaStore);
+    final Map<String, Object> props = new HashMap<>();
+    EasyMock.expect(engine.buildMultipleQueries("CS", ksqlConfig, props))
+        .andReturn(Collections.emptyList());
+    EasyMock.expect(engine.buildMultipleQueries("CT", ksqlConfig, props))
```

Comment:
Ditto here. Try and make the test focus only on what is necessary. Also, to make it readable, it is worth extracting expectations into methods with descriptive names